### ISSUE TITLE
Allow editing slider values as text, at least for DynamicMap

### ISSUE THREAD

**Issue by jbednar** [ioam/holoviews#673]
For a bounded DynamicMap, e.g. this cell from the DynamicMap tutorial:

![image](https://cloud.githubusercontent.com/assets/1695496/15184207/de23dd68-1759-11e6-8ad2-2bbd93bbf214.png)

the slider values can be set only to the precision supported by the slider, which is a couple of digits when dragging the slider, and (for this example at least) a precision of 0.001 when using the left and right arrow keys to adjust the value more precisely.  It would be great if we could click in the text box and type an arbitrarily precise value, since the underlying object supports any such value.

Obviously, such a feature would require error checking, such as cropping the value to the bounds of the range, but that doesn't seem onerous to implement.  It makes the most sense for a DynamicMap, because it's the only way I can see to provide arbitrarily precise values, but it could also be implemented for a regular HoloMap by snapping the value to the nearest defined value.

**Comment by vascotenner** [comment]
It would be very cool if values can be entered outside the range of the param, and that than the param range would increase. Often, when I investigate some parameter space, I find halfway that my ranges are insufficient. It is quite a work to create a new dmap with a larger range, set the other x parameters back to their values.

**Comment by jbednar** [comment]
Makes sense.  Sounds like you want something like the param package's "softbounds".  

ipywidgets sliders now allow editing the textual value; not sure if that's any help when trying to implement this.

**Comment by jakirkham** [comment]
As noted in issue ( https://github.com/ioam/holoviews/issues/2178 ), would be nice to see this in the context of [Gridded Datasets]( http://holoviews.org/getting_started/Gridded_Datasets.html ) as well. Still new to holoviews, so sorry if I'm just repeating what's in the OP.
